### ISSUE TITLE
modify yolov4 for early stopping

### ISSUE THREAD

**Issue by pisenk** [AlexeyAB/darknet#8808]
hi @AlexeyAB , i have been working on some projects using our yolov4 and yolov4-tiny model. My model is used for six class maturity detection of oil palm fruit and its achieved a very good mAP around 99 - 100 %. 

![chart_yolov4-tiny-TBS](https://github.com/AlexeyAB/darknet/assets/137201152/07d60c8c-066f-4004-8b33-6a9bf4adb7d5)
![chart_yolov4-TBS](https://github.com/AlexeyAB/darknet/assets/137201152/da1eafdf-6d25-44b1-b31b-9647cf30fb3f)

So based on your recommendation for six classes, in the cfg files max batches parameter would be 12,000. and i found out that both of my model convergence below 8000 iterations, and currently i want to implement genetic algorithms to finding optimal learning rate hyperparameters to get faster convergeence (below 8000 iterations). It comes to my mind to modify the yolov4 model to use early stopping for minimizing the network cost but i try to find the same issue in your repository and cannot find any reference about it.

may you help me for this case? what did i need to do? i read somes related issue, it is pointing to modify the script?

thanks in advance
